frontend/src/components/robotHardware/index.js

```diff
@@ -0,0 +1,150 @@
+      borderColor: theme.palette.hr.batteryLevel,
+      margin: 0,
+      position: 'absolute',
+      top: '3px',
+      left,
+      marginLeft: '-1px',
+    }}
+  />
+})
+
+const Battery = observer(() => {
+  const theme = useTheme()
+  // TODO: define logic for percentage in a store (data, requests to DB, etc.)
+  const percentage = 80
+  
+  let color = theme.palette.success.main
+  let amount = 'High'
+
+  if (percentage < 20) {
+    color = theme.palette.error.main
+    amount = 'Low'
+  } else if (percentage < 40) {
+    color = theme.palette.warning.main
+    amount = 'Middle'
+  } else if (percentage < 80) {
+    color = theme.palette.info.main
+    amount = 'Average'
+  }
+
+  return (
+    <Grid>
+      <Grid container justifyContent='space-between' >
+        <WidgetTitle styles={{marginBottom: '14px', fontWeight: theme.typography.fontWeightMedium}}>{amount}</WidgetTitle>
+        <WidgetTitle styles={{marginBottom: '14px', fontWeight: theme.typography.fontWeightMedium}}>{percentage}%</WidgetTitle>
+      </Grid>
+      <Grid
+        sx={{
+          height: '18px',
+          borderRadius: '6px',
+          backgroundColor: theme.palette.background.batteryLevel,
+          position: 'relative',
+          marginBottom: '40px',
+          overflow: 'hidden',
+        }}
+      >
+        <BatteryLevelDivider left='20%'/>
+        <BatteryLevelDivider left='40%'/>
+        <BatteryLevelDivider left='60%'/>
+        <BatteryLevelDivider left='80%'/>
+        <Grid sx={{
+          position: 'absolute',
+          top: 0,
+          left: 0,
+          height: '100%',
+          width: `${percentage}%`,
+          backgroundColor: color,
+          borderRadius: '6px',
+        }}/>
+      </Grid>
+    </Grid>
+  )
+})
+
+const CircleProgress = observer(({percentage, title}) => {
+  const theme = useTheme()
+
+  return (
+    <CircularProgressbarWithChildren
+      value={percentage}
+      strokeWidth={10}
+      styles={buildStyles({
+        pathColor: theme.palette.info.main,
+        trailColor: theme.palette.background.circleProgressTrail,
+      })}
+    >
+      <Typography
+        sx={{
+          color: theme.palette.primary.main,
+          lineHeight: 1,
+          fontSize: '18px',
+          fontWeight: theme.typography.fontWeightBold,
+          marginBottom: '3px',
+        }}
+      >{percentage} %</Typography>
+      <Typography
+        sx={{
+          color: theme.palette.primary.main,
+          fontSize: '14px',
+          lineHeight: 1,
+          fontWeight: theme.typography.fontWeightRegular,
+        }}
+      >{title}</Typography>
+    </CircularProgressbarWithChildren>
+  )
+})
+
+const RobotHardware = observer(() => {
+  // TODO: define logic for memoryUsagePercentage and cpuUsagePercentage in a store (data, requests to DB, etc.)
+  const memoryUsagePercentage = 35
+  const cpuUsagePercentage = 60
+
+  return (
+    <Widget widgetId='2' styles={{height: '100%'}}>
```

Comment:
Would these ids be always hardcoded?

Reply:
prop changed to `widgetName`
yes, this prop will be hardcoded